arrows/vxl/hashed_image_classifier.h

```diff
@@ -0,0 +1,186 @@
+  std::vector< weight_t > weights;
+
+  // Pointers to the start of each individual feature weight vector
+  std::vector< weight_t* > feature_weights;
+
+  // Is the current classifier configured correctly?
+  bool is_valid() const;
+
+  // Copy operator
+  self_t& operator=( self_t const& other );
+
+  // Seed the model with empty values of the given size
+  void reset( unsigned feature_count, unsigned entries_per_feature );
+
+  // Normalize internal histogram weights to a given absolute sum
+  void normalize( weight_t total_weight = 1.0 );
+
+private:
+  vital::logger_handle_t logger =
+    kwiver::vital::get_logger( "arrows.vxl.hashed_image_classifier" );
+};
+
+/// Stream operator declaration for the hashed_image_classifier model class.
+template < typename FloatType > std::ostream&
+operator<<( std::ostream& os,
+            hashed_image_classifier_model< FloatType > const& obj );
+
+/// Stream operator declaration for the hashed_image_classifier class.
+template < typename FeatureType, typename OutputType >
+std::ostream&
+operator<<( std::ostream& os,
+            hashed_image_classifier< FeatureType, OutputType > const& obj );
+
+/// A classifier designed to efficiently classify every pixel in an image.
+///
+/// Every feature (for every pixel) is required to be mapped onto unsigned
+/// integral type HashType before input. For every possible value of this
+/// hashed mapping, and for every feature, a single weight must be given (in a
+/// model file) which contributes to some binary decision about said pixel.
+///
+/// More formally, for every feature at location i,j, i.e. f(i,j,0), f(i,j,1)
+/// f(i,j,...) the output classification is given as:
+///
+///                      c(i,j) = sum( w_{k}[ f(i,j,k) ] )
+///
+///       for every feature k, and some corresponding weight vector w_{k}
+///
+/// Typically these models may be specified manually, created with the help of
+/// constructs such as parzen window estimates, or learned via supervised
+/// learning techniques such as using boosted linear discriminants. Compared
+/// against other modern classifiers, it places significant constraints on how
+/// features can be used together in favor of efficiency.
+
+template < typename FeatureType, typename OutputType = double >
+class hashed_image_classifier
+{
+public:
+  using weight_t = OutputType;
+  using weight_image_t = vil_image_view< weight_t >;
+  using mask_image_t = vil_image_view< bool >;
+  using input_t = FeatureType;
+  using input_image_t = vil_image_view< input_t >;
+  using feature_vector_t = std::vector< input_image_t >;
+  using model_t = hashed_image_classifier_model< OutputType >;
+  using model_sptr_t = std::shared_ptr< model_t >;
+  using self_t = hashed_image_classifier< FeatureType, OutputType >;
+
+  /// Default constructor; a model must be loaded via load_from_file before
+  /// use.
+  hashed_image_classifier() : model_( new model_t() ) {}
+
+  /// Descructor
```

Comment:
```suggestion
```
The style guide advises to elide "gratuitous" documentation. Anyway, this was spelled wrong and missing a period. (Apologies for missing this in the previous review.)